docs/contracts/v4/first-hook/03-hook-deployment.mdx

```diff
@@ -0,0 +1,122 @@
+   you have the option to deploy it locally on your machine if required.
+
+2. **Deploying the Hook Contract**: The hook contract needs to be deployed at a predetermined address. You can use
+   `CREATE2` for deterministic deployment. A Deterministic Deployment Proxy, usually found
+   at `0x4e59b44847b379578588920cA78FbF26c0B4956C`, is employed for this purpose and is already available in most
+   environments.
+
+3. **Deploying Test Tokens**: These tokens are essential for creating the pool. They need to be deployed before
+   initializing the pool.
+
+4. **Initializing the Pool with the Hook Contract Address**: This is achieved by invoking
+   the `initialize(PoolKey memory key, uint160 sqrtPriceX96, bytes calldata hookData)` function on the PoolManager contract.
+
+5. **Adding Liquidity or Modifying Position**: If you wish to add liquidity to the pool or alter its position, a
+   utility contract that implements the `ILockCallback` interface is necessary. You may consider deploying a utility
+   contract like `PoolModifyPositionTest` for these operations.
+
+
+## Deployment Scripts
+The template includes a few scripts that help with deploying hooks. These scripts are located in the `scripts` folder.
+
+Lets look at these scripts one by one:
+
+### 1. Deploying Your Own Tokens
+The template includes Mock UNI and Mock USDC contracts for testing. Deploy them using:
+```shell
+ forge create script/mocks/mUNI.sol:MockUNI \
+ --rpc-url [your_rpc_url_here] \
+ --private-key [your_private_key_on_goerli_here] \
+
+ forge create script/mocks/mUSDC.sol:MockUSDC \
+ --rpc-url [your_rpc_url_here] \
+ --private-key [your_private_key_on_goerli_here] \
+```
+
+
+### 2. script/PoolCreateSolo.s.sol
+This script contains the steps for initializing the pool without a hook. It uses the pre-deployed PoolManager contract and
+token contracts
+
+```solidity
+address constant GOERLI_POOLMANAGER = address(0x3A9D48AB9751398BbFa63ad67599Bb04e4BdF98b); // pool manager deployed to GOERLI
+address constant MUNI_ADDRESS = address(0xbD97BF168FA913607b996fab823F88610DCF7737); // mUNI deployed to GOERLI -- insert your own contract address here
+address constant MUSDC_ADDRESS = address(0xa468864e673a807572598AB6208E49323484c6bF); // mUSDC deployed to GOERLI -- insert your own contract address here
+address constant HOOK_ADDRESS = address(0x0); // hookless pool is 0x0!
+```
+Here it uses the pre-deployed PoolManager contract and token contracts. You can replace these addresses with your own.
+
+It then initialized the pool without the hook
+
+```solidity
+bytes memory hookData = new bytes(0);
+
+PoolKey memory pool = PoolKey({
+    currency0: Currency.wrap(token0),
+    currency1: Currency.wrap(token1),
+    fee: swapFee,
+    tickSpacing: tickSpacing,
+    hooks: IHooks(address(HOOK_ADDRESS)) // 0x0 is the hookless pool
+});
+
+manager.initialize(pool, startingPrice, hookData);
+```
```

Comment:
I dont know if we need to document the hookless pool creation. I dont think its relevant for devs looking to deploy their first hook